### PR TITLE
Bug 1493942 - Support SNAPSHOT artifacts

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Depends on https://github.com/mozilla-releng/mozilla-version/pull/44

### CODE REVIEW

**Comment by JohanLorenzo** [comment]
I tested this function under mozilla-releng's `populateMavenMetada` lambda function. I didn't get any error nor noticed any weird behavior. The next step is to end-to-end test android-component's SNAPSHOT uploads.

File: maven_lambda/__init__.py [hunk 1]
```diff
@@ -108,14 +143,97 @@ def generate_release_maven_metadata(folder_content_keys):
...
-        stream, encoding='unicode', xml_declaration=True, method='xml', short_empty_elements=True
+        stream, encoding='unicode', xml_declaration=True, method='xml', short_empty_elements=False
     )
     return stream.getvalue()
 
 
+def generate_snapshot_listing_metadata(bucket_name, pom_files_keys):
+    snapshots_metadata = get_snapshots_metadata(bucket_name, pom_files_keys)
+    sorted_snapshot_metadata = sorted(
+        snapshots_metadata, key=lambda x: x['build_number'], reverse=True
```
Comment (MihaiTabara) [inline maven_lambda/__init__.py]: Okay, so we're sticking to 32bit-integer here for buildnumbers for now 👍

File: maven_lambda/__init__.py [hunk 2]
```diff
@@ -145,3 +267,7 @@ def generate_checksums(data):
         'md5': hashlib.md5(data).hexdigest(),
         'sha1': hashlib.sha1(data).hexdigest(),
     }
+
+
+def is_snapshot(key):
+    return '-SNAPSHOT/' in key
```
Comment (MihaiTabara) [inline maven_lambda/__init__.py]: Not sure if the trailing backslash makes sense to be in the check. It doesn't hurt for sure, but thought it's cleaner in its absence.

Comment (JohanLorenzo) [reply maven_lambda/__init__.py]: Okay. Done!

**Pull request merged**